en/advanced/parameters_and_configurations.md

```diff
@@ -29,52 +31,189 @@ x   RC_MAP_ACRO_SW [375,514] : 0
  723 parameters total, 532 used.
 ```
 
-### Exporting and Loading Parameters
+You can use the `-c` flag to show all parameters that have changed (from their defaults):
+```sh
+param show -c
+```
 
-The standard save command will store the parameters in the current default file:
+### Exporting and Loading Parameters
 
+The standard `param save` command will store the parameters in the current default file:
 ```sh
 param save
 ```
 
 If provided with an argument, it will store the parameters instead to this new location:
-
 ```sh
 param save /fs/microsd/vtol_param_backup
 ```
 
-There are two different commands to load parameters: `param load` will load a file and replace the current parameters with what is in this file, resulting in a 1:1 copy of the state that was present when these parameters were stored. `param import` is more subtle: It will only change parameter values which have been changed from the default. This is great to e.g. initially calibrate a board (without configuring it further), then importing the calibration data without overwriting the rest of the system configuration.
-
-Overwrite the current parameters:
-
+There are two different commands to load parameters: 
+- `param load` replaces all the current parameters, duplicating the state when the parameters were stored.
+   parameters are not automatically saved by this operation (it is assumed they are already saved).
+- `param import` only updates parameter values that have been *changed from the default*.
```

Comment:
Hi @bkueng That comment makes no sense to me at all - ie I can't see how that differs from `load`.

Let me see if I an deconstruct it ...

> A params file typically contains all parameters that an aifrframe config or the user touched/changed (also if changed back to default).

So basically a param file just contains _changed_ parameters. Presumably this means there is something in the system that notices something has been changed (as opposed to just comparing to a stored default value).

So when this is imported it will only update the parameters it contains (things that have been changed, even if they were changed back to defaults) . 

So import replaces all values. If a value was changed it will be updated even if it was changed back to default. 
1. What implication does that have? If it was originally a default value and it gets set to that same default value isn't that the same thing as not having changed the value in the first place. And if not, why? 
2.  The possible implication I can see is if the default value in file differs from the default value on vehicle then that parameter will be updated. But presumably the value will not be marked as a "default" even though it was in the original
3. Still can't see how import and load actually differ!


How does that differ from `load`? The only thing

Reply:
> What implication does that have? If it was originally a default value and it gets set to that same default value isn't that the same thing as not having changed the value in the first place. And if not, why?

There's a subtle case where it's not the same: if the default value in the Firmware changes, you will keep your 'default' if you ever set the param, and if not, you will use the updated Firmware default when flashing the new version.

> The possible implication I can see is if the default value in file differs from the default value on vehicle then that parameter will be updated. But presumably the value will not be marked as a "default" even though it was in the original

Maybe this helps: a params file does not have any notion of default. It just stores a list of param names with an associated value. All default/untouched params are simply not stored in that file.

> Still can't see how import and load actually differ!

You can see it as following: a `load` does a full reset to default first for all params, and then imports the params.

Reply:
@bkueng A real challenge, but I now fully get this. Please check the updated version (the whole section updated).